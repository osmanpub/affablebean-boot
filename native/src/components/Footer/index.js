import React from "react";
import { Button, StyleSheet, Text, View } from "react-native";

const styles = StyleSheet.create({
  Button: {
    color: "red"
  }
});

export default function Footer(props) {
  function onPressContact() {}

  function onPressPrivacy() {}

  return (
    <View
      style={{
        alignItems: "center",
        flex: 1,
        justifyContent: "center"
      }}
    >
      <View
        style={{
          alignItems: "center",
          flexDirection: "row"
        }}
      >
        <Button
          accessibilityLabel="Privacy"
          onPress={onPressPrivacy}
          style={styles.Button}
          title="Privacy"
        />
        <View style={{ flex: 0.1 }} />
        <Button
          accessibilityLabel="Contact us"
          onPress={onPressContact}
          style={styles.Button}
          title="Contact us"
        />
      </View>
      <Text>&copy; 2019 the affable bean company</Text>
    </View>
  );
}
