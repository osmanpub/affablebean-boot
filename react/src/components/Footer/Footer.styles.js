import styled from "styled-components";

export const FooterWrapper = styled.div`
  margin: 0 110px;
  height: 60px;
  width: 800px;
  clear: left;
`;

export const FooterDivider = styled.hr`
  margin-left: 300px;
  width: 600px;
`;

export const FooterText = styled.span`
  margin-left: 110px;
  margin-top: 5px;
  width: 600px;
`;
