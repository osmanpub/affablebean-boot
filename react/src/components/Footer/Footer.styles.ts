import styled from "styled-components";

export const FooterWrapper = styled.div`
  margin: 0 110px;
  height: 60px;
  width: 800px;
  clear: left;
`;

export const FooterText = styled.span`
  margin-left: 110px;
  margin-top: 5px;
  width: 600px;
`;
