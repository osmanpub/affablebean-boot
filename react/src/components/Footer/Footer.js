import React from "react";
import { FooterDivider, FooterWrapper } from "./Footer.styles";

export function Footer(props) {
  return (
    <FooterWrapper>
      <div>
        <br />
        <FooterDivider />
        <p>
          <button>Privacy</button>
          <button>Contact</button>
          &nbsp;&copy;&nbsp;2019 the affable bean company
        </p>
      </div>
    </FooterWrapper>
  );
}
