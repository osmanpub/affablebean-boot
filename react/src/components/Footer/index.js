import React from "react";
import { Footer } from "./Footer";

export default React.memo(Footer);
