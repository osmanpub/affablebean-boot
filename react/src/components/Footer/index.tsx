import React from "react";
import { Link } from "react-router-dom";
import { FooterText, FooterWrapper } from "./Footer.styles";

export default function Footer() {
  return (
    <FooterWrapper>
      <br />
      <FooterText>
        <Link to={"/privacy"} data-cy="privacy" data-testid="footer-privacy">
          <button className="btn btn-link">Privacy</button>
        </Link>
        <Link to={"/contact"} data-cy="contact" data-testid="footer-contact">
          <button className="btn btn-link">Contact</button>
        </Link>
        &nbsp;&copy;&nbsp;2020 the affable bean company
      </FooterText>
    </FooterWrapper>
  );
}
