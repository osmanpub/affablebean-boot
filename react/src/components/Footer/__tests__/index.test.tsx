import React from "react";
import { BrowserRouter as Router } from "react-router-dom";
import renderer from "react-test-renderer";
import Footer from "..";

describe("<Footer />", () => {
  it("renders correctly", () => {
    const component = renderer
      .create(
        <Router>
          <Footer />
        </Router>
      )
      .toJSON();
    expect(component).toMatchSnapshot();
  });
});
