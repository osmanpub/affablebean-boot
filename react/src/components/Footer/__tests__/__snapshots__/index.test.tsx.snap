// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> renders correctly 1`] = `
<div
  className="sc-AykKD crBykC"
>
  <br />
  <span
    className="sc-AykKE fYjhKo"
  >
    <a
      data-cy="privacy"
      href="/privacy"
      onClick={[Function]}
    >
      <button
        className="btn btn-link"
      >
        Privacy
      </button>
    </a>
    <a
      data-cy="contact"
      href="/contact"
      onClick={[Function]}
    >
      <button
        className="btn btn-link"
      >
        Contact
      </button>
    </a>
     © 2019 the affable bean company
  </span>
</div>
`;
